cpp/src/arrow/compute/api_vector.cc

```diff
@@ -325,6 +342,15 @@ Result<std::shared_ptr<Array>> DropNull(const Array& values, ExecContext* ctx) {
   return out.make_array();
 }
 
+// ----------------------------------------------------------------------
+// Cumulative functions
+
+Result<Datum> CumulativeSum(const Datum& values, const CumulativeSumOptions& options,
+                            ExecContext* ctx) {
+  auto func_name = (options.check_overflow) ? "cumulative_sum_checked" : "cumulative_sum";
+  return CallFunction(func_name, {Datum(values)}, &options, ctx);
```

Comment:
I find this API design a bit weird. What happens if I call "cumulative_sum_checked" but pass `check_overflow` set to false in the options struct (or conversely, call "cumulative_sum" with `check_overflow` set to true)?

Reply:
Probably remove the `check_overflow` option and expose the two variants as separate functions in the public API.
@pitrou Do you have any suggestions on how to resolve this?